### ISSUE TITLE
Consider placing remote functionality behind a feature flag

### ISSUE THREAD

**Issue by outis151** [bgpkit/bgpkit-parser#181]
For applications where only local MRT files are to be processed it would be nice to be able to remove the need to compile all the 
 components for fetching remote MRT files. I tried doing this by only using the `parser` feature flag, however this leads to an error with `oneio` being required for `OneIoError`.

**Comment by digizeph** [comment]
Thank you for your suggestion!

I have put together PR #183 to resolve this. Could you take a look?

I have also added an example that uses only the `parser` feature and it should not bring in any remote-related dependencies.

If you have the source code checked out, you can run this to check the dependencies yourself.
```
git fetch --all
git checkout feature/local_only
cd examples/local_only
cargo tree
```

**Comment by digizeph** [comment]
Thanks for testing! I have merged #183 to main branch. Will cut a release soon, perhaps after I figure out a plan for #182.

**Comment by digizeph** [comment]
@outis151 i've added another patch that further decouple oneio from the parser code: 4b77766087167ca7b2816b427be19630f52ba12a

After this commit, when using no default and `parser` feature, the library will not bring `oneio` dependency in, and you can pass in your own reader that implements `Read+Send` via the `BgpkitParser::new_with_reader(reader: Box<dyn Read + Send>)` function. The previously default `BgpkitParser::new(path: &str)` is now behind oneio dependency. You can get use `parser,oneio` features to get the dependencies for local-only oneio without any remote functionalities, as in #183.
